documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,170 @@
+
+### 3.3 Redirect map with fast path
+The Fast Path here is not ER Fast Path, but the fast path that required by PLS. Fast path ICMP flow redirection will still take effect when redirect map is used.
+
+- If PL NSG is not used, it changes the flow just like regular PL case.
+
+- If PL NSG is used, it only updates the PL encap, leaving the NSG encap unchanged.
+
+For more information on how Fast Path works, please refer to [Fast Path documentations](../load-bal-service/fast-path-icmp-flow-redirection.md).
+
+### 3.4 Non-required features 
+- RST on connection idle timeout.
+- Flow resimulation should not be triggered automatically on redirect map update.
+
+## 4. Resource modeling, requirement, and SLA
+### 4.1 Scaling requirement
+For scaling requirement, please refer to [SONIC DASH HLD](https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md#14-scaling-requirements).
+
+### 4.2 Reliability requirements
+The flows created when redirect map will be replicated among multiple DPUs, following the SmartSwitch HA design.
+
+For more information, please refer to [SmartSwitch HA design doc](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md).
+
+## 5. SAI API design
+### 5.1. DASH Outbound Port Map
+A new object outbound-port-map is added, which aims to define SNAT/DNAT port translation in outbound direction. It includes the following attributes:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_PORT_MAP_ATTR_COUNTER_ID | sai_object_id_t | Attach a counter. |
+
+NOTE: More attributes could be added in the future for SNAT/DNAT scenarios.
+
+### 5.2. DASH Outbound Port Map Port Range
+A new object key entry outbound-port-map-port-range is added, which aims to define port range translation in outbound direction. It is suitable for setting up the PL / PL NSG redirect map scenarios.
+
+The entry is defined like below:
+```c
+typedef struct _sai_outbound_port_map_port_range_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Exact matched key outbound_port_map_id
+     *
+     * @objects SAI_OBJECT_TYPE_OUTBOUND_PORT_MAP
+     */
+    sai_object_id_t outbound_port_map_id;
+
+    /**
+     * @brief Range matched key dst_port_range
+     */
+    sai_u32_range_t dst_port_range;
+
+    /**
+     * @brief Rule priority in table
+     */
+    sai_uint32_t priority;
+
+} sai_outbound_port_map_port_range_entry_t;
+```
+
+ It includes the following attributes:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_ACTION | sai_outbound_port_map_port_range_entry_action_t | Action. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_IP | sai_ip_address_t | Back end IP. |
```

Comment:
this IP also need to rewrite the underlay dip beside being used as the inner packet dip in 4to6 action.